lib/http/base.js

```diff
@@ -93,27 +96,34 @@ HttpClient.prototype.request = function (options)
       mock);
 
     requestOptions.data = requestOptions.body;
-    requestOptions.httpsAgent = agentAndProxyOptions.agentClass(agentAndProxyOptions.agentOptions);
-    requestOptions.httpsAgent.keepAlive = agentAndProxyOptions.agentOptions.keepAlive;
+    requestOptions.dispatcher = agentAndProxyOptions.agentClass();
     requestOptions.retryDelay = this.constructExponentialBackoffStrategy();
+    // for urllib2 we set keepAlive flag but for urllib is true by default
 
-    request = require('urllib').request(requestOptions.url, requestOptions, async function (err, data, response)
-    {
-      // if a callback has been specified, normalize the
-      // response before passing it to the callback
-      if (Util.isFunction(options.callback))
-      {
-        if (err)
-        {
+    if (Util.isFunction(options.callback)) {
+      urllib.request(requestOptions.url, requestOptions)
+        .then(async function ({ data, res }) {
+          if (res.headers['x-goog-stored-content-encoding'] === 'GZIP') { // Google API sets own header and urllib cannot parse it automatically
```

Comment:
I believe headers may be a list of values and you compare with single one?

Reply:
Here we expect it to be only GZIP

Reply:
Yes, but does this comparison compares list of strings with simple string? Is the type of `res.headers['x-goog-stored-content-encoding']` is string or list of strings?

Reply:
- it's response, not a content negotiation
- we expect only one content encoding here
https://cloud.google.com/storage/docs/xml-api/reference-headers#xgoogstoredcontentencoding